src/macos/PolicyWatcher.cc

```diff
@@ -80,15 +80,25 @@ void PolicyWatcher::Execute(const ExecutionProgress &progress)
     while (!disposed)
     {
         updatedPolicies.clear();
+        bool update = false;
         for (auto &policy : policies)
-        {
-            if (policy->refresh())
+        {            
+            switch(policy->refresh())
             {
-                updatedPolicies.push_back(policy.get());
+                case PolicyRefreshResult::Updated:
+                    updatedPolicies.push_back(policy.get());
+                    update = true;
+                    break;
+                case PolicyRefreshResult::Unchanged:
+                    updatedPolicies.push_back(policy.get());
+                    break;
+                case PolicyRefreshResult::Removed:
+                    update = true;
+                    break;
             }
         }
 
-        if (first || updatedPolicies.size() > 0)    
+        if (first || update)
             progress.Send(&updatedPolicies[0], updatedPolicies.size());
```

Comment:
What happens if updatedPolicies size is zero?

Reply:
The effect is that the policy gets cleared

Reply:
Is there a chance that line 102 gets executed even if updatePolicies size is zero? It seems not obvious.

Reply:
It is intentional that line 102 gets executed if updatedPolicies is 0.  Otherwise, I don't see a way to "clear" policy that has been removed by the user.